### PR TITLE
usaco-807: CPP Solution added

### PR DESCRIPTION
**Pull request by Thamognya**
- USACO 807 CPP Solution added

<hr>

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ x ] I have tested my code.
- [ x ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ x ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-807.mdx [hunk 1]
```diff
@@ -74,4 +74,38 @@ public class Teleportation {
...
+    freopen ("teleport.out", "w", stdout);
+
+	// define variables needed; i is the going to be input array
+	// j is going to be best strategy's distance required
+    int i[4], j;
+    cin >> i[0] >> i[1] >> i[2] >> i[3];
+
+    j = abs(i[0] - i[1]); // caluculate the distance without travel
+	// if it is greater than the sum with teleportation then change it to sum with teleportation
+    if (j > abs(i[0] - i[2]) + abs(i[1] - i[3])) {
```
Comment (envyaims) [inline solutions/usaco-807.mdx:97]: use min function

File: solutions/usaco-807.mdx [hunk 2]
```diff
@@ -74,4 +74,38 @@ public class Teleportation {
...
+        j = abs(i[0] - i[2]) + abs(i[1] - i[3]);
+    }
+    // repeat this again with the other possible method
```
Comment (envyaims) [inline solutions/usaco-807.mdx:100]: well, what is the other possible method?

File: solutions/usaco-807.mdx [hunk 3]
```diff
@@ -74,4 +74,38 @@ public class Teleportation {
...
+    freopen ("teleport.in", "r", stdin);
+    freopen ("teleport.out", "w", stdout);
+
+	// define variables needed; i is the going to be input array
+	// j is going to be best strategy's distance required
+    int i[4], j;
+    cin >> i[0] >> i[1] >> i[2] >> i[3];
+
+    j = abs(i[0] - i[1]); // caluculate the distance without travel
+	// if it is greater than the sum with teleportation then change it to sum with teleportation
```
Comment (envyaims) [inline solutions/usaco-807.mdx:96]: unnecessary
```suggestion

```

File: solutions/usaco-807.mdx [hunk 4]
```diff
@@ -74,4 +74,38 @@ public class Teleportation {
 	
 </JavaSection>
 
+<CPPSection>
+
+```cpp
+#include <iostream>
```
Comment (envyaims) [inline solutions/usaco-807.mdx:80]: just use `bits/stdc++.h` lol

**Review by envyaims — CHANGES_REQUESTED** [review]

**Comment by envyaims** [comment]
also dont forget to add yourself to the authors list :)

File: solutions/usaco-807.mdx [hunk 5]
```diff
@@ -74,4 +74,38 @@ public class Teleportation {
...
+#include <stdio.h>
+#include <stdlib.h>
+using namespace std;
+
+int main(void) {
```
Comment (envyaims) [inline solutions/usaco-807.mdx:85]: this isn't c
```suggestion
int main() {
```

File: solutions/usaco-807.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-807.mdx:80]: (optional)

File: solutions/usaco-807.mdx [hunk 6]
```diff
@@ -74,4 +74,38 @@ public class Teleportation {
...
+    if (j > abs(i[0] - i[2]) + abs(i[1] - i[3])) {
+        j = abs(i[0] - i[2]) + abs(i[1] - i[3]);
+    }
+    // repeat this again with the other possible method
+    if (j > abs(i[0] - i[3]) + abs(i[1] - i[2])) {
+        j = abs(i[0] - i[3]) + abs(i[1] - i[2]);
+    }
+
+	// print the solution to teleport.out
+    cout << j << "\n";
```
Comment (SansPapyrus683) [inline solutions/usaco-807.mdx:106]:
```suggestion
    cout << j << endl;
```

File: solutions/usaco-807.mdx (hunk 5 above)
Comment (Thamognya) [reply solutions/usaco-807.mdx:85]: ah sorry about that. I usually code a lot in c.

**Comment by Thamognya** [comment]
> also dont forget to add yourself to the authors list :)

Sorry for my redundant commenting, but could I ask where it might be in the repository.

**Comment by envyaims** [comment]
> > also dont forget to add yourself to the authors list :)
> 
> 
> 
> Sorry for my redundant commenting, but could I ask where it might be in the repository.

at the top of the mdx file?

File: solutions/usaco-807.mdx [hunk 7]
```diff
@@ -74,4 +74,33 @@ public class Teleportation {
...
+
+int main() {
+	freopen ("teleport.in", "r", stdin);
+	freopen ("teleport.out", "w", stdout);
+
+	// define variables needed; arr is the going to be input array
+	int arr[4];
+	cin >> arr[0] >> arr[1] >> arr[2] >> arr[3];
+
+	//  dist is going to be best strategy's distance required
```
Comment (envyaims) [inline solutions/usaco-807.mdx:91]: what does this mean

**Comment by envyaims** [comment]
> indentation is still kinda weird

bruh i fixed it and he unfixed it

**Review by envyaims — CHANGES_REQUESTED** [review]
i think you should use a,b,x,y like the other solutions and the actual statement. You reading into an array might confuse beginners.

**Review by envyaims — APPROVED** [review]
good enough lol

**Pull request merged**